[flink] Document extra_metrics for custom/user metrics (#24741)
DataDog/integrations-core: flink/README.md

3. Restart Flink and the Agent.

Flink jobs can register their own [custom/user metrics][16] (for example, an
operator-scoped counter your operator code emits). By default, this
integration only collects the metrics documented in [metadata.csv][11] --
job-specific custom metrics are not collected out of the box. Collect them
with the `extra_metrics` option, mapping the exact raw Prometheus metric name

> Flink exposes to the Datadog metric name you want:

Reviewer: Four small things in this paragraph:

- `custom/user metrics` uses `/` to mean "and/or", which Datadog style avoids. It's also a terminology inconsistency. The term appears once here and the rest of the block consistently says "custom metrics" ("job-specific custom metrics", "a custom metric's exposed type", "These custom Flink metrics").
- ` -- ` is a double hyphen standing in for an em dash.
- `only collects the metrics documented` — misplaced modifier; `only` should sit directly before what it modifies.
- `By default, this integration only collects...` and `are not collected out of the box` say the same thing twice in one sentence. "Out of the box" is also colloquial.

```suggestion
Flink jobs can register their own [custom metrics][16] (for example, an
operator-scoped counter your operator code emits). By default, this
integration collects only the metrics documented in [metadata.csv][11], so
job-specific custom metrics are not collected. Collect them
with the `extra_metrics` option, mapping the exact raw Prometheus metric name
Flink exposes to the Datadog metric name you want:
```

```yaml
instances:
  - openmetrics_endpoint: http://<FLINK_HOST>:9249/metrics
    extra_metrics:
      - flink_taskmanager_job_task_operator_messageLatency: operator.messageLatency
```

> <div class="alert alert-info">These custom Flink metrics are considered standard metrics in Datadog.</div>

Reviewer: Confirming the precedent you flagged for @steveny91, since it's easy to verify and saves them a step: this construction is well established.

- `torchserve/README.md:80` — `<div class="alert alert-info">These custom TorchServe metrics are considered standard metrics in Datadog.</div>` (identical construction and alert class)
- `nvidia_triton/README.md:52` — same sentence, but `alert-warning`
- `ray/README.md:92` — same sentence as `**Note:**`

So the wording and the `alert-info` class both match existing practice, and the three existing integrations already disagree on presentation, so that inconsistency predates this PR. No editorial change needed.

The *billing* accuracy of the claim is still a question only @steveny91 can answer. I'm only confirming the wording precedent.

> You can also match multiple metrics at once with a regular expression, for
> example to collect everything under a given Flink scope:

Reviewer: Missing comma after `for example`

```suggestion
You can also match multiple metrics at once with a regular expression, for
example, to collect everything under a given Flink scope:
```

```yaml

>     extra_metrics:

Reviewer: Optional: this block and the one at line 96 both begin with a 4-space-indented `extra_metrics:` and no parent key, while the first block at line 71 shows the full `instances:` context. A reader copying from the middle of the page won't see where the key nests. Worth considering whether to show full context in each block, or dedent these two. Not wrong as-is.

      - ^flink_taskmanager_job_task_operator_.+
```

Metrics collected this way are submitted under Flink's raw Prometheus metric
name (for example `flink.flink_taskmanager_job_task_operator_messageLatency`)
rather than a curated Datadog name, since a regular expression can match more

> than one metric and therefore can't be mapped to a single custom name.

Reviewer: This sentence runs about 45 words, well over the 25-word guideline. Splitting it also drops the `can't` contraction, which is out of step with the rest of the file ("Flink does not include any service checks").

```suggestion
Metrics collected this way are submitted under Flink's raw Prometheus metric
name (for example `flink.flink_taskmanager_job_task_operator_messageLatency`)
rather than a curated Datadog name. A regular expression can match more than
one metric, so it cannot be mapped to a single custom name.
```
